napari/_tests/test_viewer.py

```diff
@@ -135,5 +135,58 @@ def test_changing_theme(viewer_factory):
     viewer.theme = 'light'
     assert viewer.palette['folder'] == 'light'
 
+
+    # Close the viewer
+    viewer.window.close()
+
     with pytest.raises(ValueError):
         viewer.theme = 'nonexistent_theme'
+        
+        
+def test_roll_traspose_update(qtbot):
+    """ Controlling that translation and rolling preserve correct sequence
+    of translate and scaling values.
+    """
+
+    def check_view_consistency(viewer, transf_dict):
+        """ Utility function for doing the checks.
+        """
+        # Get an handle on visual layer:
+        vis_lyr = viewer.window.qt_viewer.layer_to_visual[lyr1]
+
+        # Visual layer attributes should match expected from viewer dims:
+        for transf_name, transf in transf_dict.items():
+            disp_dims = viewer.dims.displayed  # dimensions displayed in 2D
+            vis_vals = getattr(vis_lyr, transf_name)[
+                1::-1
+            ]  # values of visual layer
+            correct_vals = transf[disp_dims]  # expected translate/scale values
+            assert (vis_vals == correct_vals).all()
+
+    viewer = Viewer()
+    view = viewer.window.qt_viewer
+    qtbot.addWidget(view)
```

Comment:
as described [here](https://napari.org/docs/developers/CONTRIBUTING.html#writing-tests) it's potentially problematic to _both_ add a widget to the `qtbot` and perform your own cleanup (`viewer.window.close()`) at the end of the test.  If at all possible, prefer the `viewer_factory()` to creating your own Viewer,